test/autocast_fp8.cpp

```diff
@@ -0,0 +1,142 @@
+    }
+    run_pass(m1);
+
+    migraphx::module m2;
+    {
+        auto y_fp32 = m2.add_parameter("y", {migraphx::shape::float_type, {1}});
+        auto x_fp32 = m2.add_parameter("x", {migraphx::shape::float_type, {1}});
+        auto y_fp8  = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::fp8e4m3fnuz_type}}), y_fp32);
+        auto x_fp8  = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::fp8e4m3fnuz_type}}), x_fp32);
+        m2.add_instruction(migraphx::make_op("sub"), x_fp8, y_fp8);
+    }
+    EXPECT(m1 == m2);
+}
+
+// multiple inputs to return
+TEST_CASE(autocast_fp8_3)
+{
+    migraphx::module m1;
+    {
+        auto x      = m1.add_parameter("x", {migraphx::shape::fp8e4m3fnuz_type, {1}});
+        auto y      = m1.add_parameter("y", {migraphx::shape::fp8e4m3fnuz_type, {1}});
+        auto sum    = m1.add_instruction(migraphx::make_op("add"), x, y);
+        auto diff   = m1.add_instruction(migraphx::make_op("sub"), x, y);
+        auto result = m1.add_instruction(migraphx::make_op("concat", {{"axis", 0}}), sum, diff);
+        m1.add_return({result});
+    }
+    run_pass(m1);
+
+    migraphx::module m2;
+    {
+        auto y_fp32      = m2.add_parameter("y", {migraphx::shape::float_type, {1}});
+        auto x_fp32      = m2.add_parameter("x", {migraphx::shape::float_type, {1}});
+        auto y_fp8       = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::fp8e4m3fnuz_type}}), y_fp32);
+        auto x_fp8       = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::fp8e4m3fnuz_type}}), x_fp32);
+        auto sum_fp8     = m2.add_instruction(migraphx::make_op("add"), x_fp8, y_fp8);
+        auto diff_fp8    = m2.add_instruction(migraphx::make_op("sub"), x_fp8, y_fp8);
+        auto concat_fp8  = m2.add_instruction(migraphx::make_op("concat", {{"axis", 0}}), sum_fp8, diff_fp8);
+        auto result_fp32 = m2.add_instruction(migraphx::make_op("convert", {{"target_type", migraphx::shape::float_type}}), concat_fp8);
+        m2.add_return({result_fp32});
+    }
+    EXPECT(m1 == m2);
+}
+
+// autocast pass does not do any changes
+TEST_CASE(autocast_fp8_4)
+{
+    migraphx::module m1;
+    {
+        auto x   = m1.add_parameter("x", {migraphx::shape::float_type, {1}});
+        auto y   = m1.add_parameter("y", {migraphx::shape::float_type, {1}});
+        auto sum = m1.add_instruction(migraphx::make_op("add"), x, y);
+        m1.add_return({sum});
+    }
+    run_pass(m1);
+
+    migraphx::module m2;
+    {
+        auto x   = m2.add_parameter("x", {migraphx::shape::float_type, {1}});
+        auto y   = m2.add_parameter("y", {migraphx::shape::float_type, {1}});
+        auto sum = m2.add_instruction(migraphx::make_op("add"), x, y);
+        m2.add_return({sum});
+
+    }
+    EXPECT(m1 == m2);
+}
+
+int main(int argc, const char* argv[]) { test::run(argc, argv); }
```

Comment:
add test with "return" statement that returns multiple instructions but each or some of them with different types.

Reply:
Added test for returning results of multiple instructions.